corehq/apps/data_interfaces/tests/test_case_deduplication.py

```diff
@@ -0,0 +1,486 @@
+from datetime import datetime
+from itertools import chain
+
+from django.test import TestCase, override_settings
+
+from dateutil.relativedelta import relativedelta
+from faker import Faker
+from mock import patch
+
+from casexml.apps.case.mock import CaseFactory
+from pillowtop.es_utils import initialize_index_and_mapping
+
+from corehq.apps.change_feed import topics
+from corehq.apps.change_feed.topics import get_multi_topic_offset
+from corehq.apps.data_interfaces.deduplication import find_duplicate_cases
+from corehq.apps.data_interfaces.models import (
+    AutomaticUpdateRule,
+    CaseDeduplicationActionDefinition,
+    CaseDeduplicationMatchTypeChoices,
+    CaseDuplicate,
+)
+from corehq.apps.data_interfaces.pillow import CaseDeduplicationProcessor
+from corehq.apps.es.tests.utils import es_test
+from corehq.elastic import get_es_new, send_to_elasticsearch
+from corehq.form_processor.interfaces.dbaccessors import CaseAccessors
+from corehq.form_processor.tests.utils import FormProcessorTestUtils
+from corehq.pillows.case_search import transform_case_for_elasticsearch
+from corehq.pillows.mappings.case_search_mapping import CASE_SEARCH_INDEX_INFO
+from corehq.pillows.xform import get_xform_pillow
+from corehq.util.elastic import ensure_index_deleted
+from corehq.util.test_utils import (
+    create_test_case,
+    flag_enabled,
+    trap_extra_setup,
+)
+
+
+class CaseRuleCriteriaTest(TestCase):
+
+    @classmethod
+    def setUpClass(cls, *args, **kwargs):
```

Comment:
since this is an entirely a SQL test you can use `setUpTestData` (which doesn't require calling super) and leave out the teardown (SQL transactions will take care of cleanup)

Reply:
Cool! I didn't know about that. Have updated.